mlflow/utils/validation.py

```diff
@@ -33,6 +33,7 @@
 MAX_ENTITY_KEY_LENGTH = 250
 MAX_MODEL_REGISTRY_TAG_KEY_LENGTH = 250
 MAX_MODEL_REGISTRY_TAG_VALUE_LENGTH = 5000
+MAX_EXPERIMENTS_LISTED_PER_PAGE = 1000
```

Comment:
This value probably should be larger than 1k. The 50k 'threshold' for search seems make sense.

Reply:
I think we should remove this client-side validation for the page length. It's up to the backend whether or not a particular `max_results` value is supported. I don't see a good reason to perform that validation here (we don't perform client-side validation for search).

I've also proposed that we update the proto docstring for the `max_results` field in SearchRuns to indicate that all servers are guaranteed to support a `max_results` threshold of at least 50,000 but may support more (see https://github.com/mlflow/mlflow/pull/3881/files#r651181532).